File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/abstractpainting/domain/AbstractPaintingObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.abstractpainting.domain;

/**
 * Created by TCC-2-9002 on 2016/10/20.
 */

public abstract class AbstractPaintingObject {
    public abstract String objAsString();
    public static AbstractPaintingObject objFromString(String str) {
        switch (str) {
            case "marker":
                return new AbstractPaintingMarkerObject();
            case "tree":
                return new AbstractPaintingTreeObject();
            default:
                return new AbstractPaintingEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/abstractpainting/domain/AbstractPaintingObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.abstractpainting.domain

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

enum class AbstractPaintingObject {
    Empty, Forbidden, Marker, Painting
}

class AbstractPaintingGameMove(val p: Position, var obj: AbstractPaintingObject = AbstractPaintingObject.Empty)

